src/middleware/overview.middleware.js

```
@@ -55,6 +56,130 @@ const fetchEntityCounts = async (req, res, next) => {
+    logger.error('Invalid noticeFlags or datasets structure')
+    next()
+  }
+
+  req.datasets = datasets.map(dataset => {
+    const notice = noticeFlags.find(notice => notice.dataset === dataset.dataset)
+
+    if (!notice || (!notice.dueNotice && !notice.overdueNotice)) {
+      return dataset
+    }
+
+    if (!(notice.deadline instanceof Date) || isNaN(notice.deadline.getTime())) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)